integration_tests/tests/e2e-tests.ts

```diff
@@ -0,0 +1,154 @@
+        data.eventTime,
+        data.transactionQty,
+        eventName
+      );
+      eventTime = data.eventTime;
+      await assertQueryResultWithTestData(
+        data,
+        dataRetrievedFromConfig.unitPrice,
+        dataRetrievedFromConfig.vendorId,
+        dataRetrievedFromConfig.serviceName
+      );
+    }
+  );
+
+  afterEach(async () => {
+    await deleteS3Events(eventIds, eventTime);
+    await deleteS3Object({
+      bucket: storageBucket,
+      key: `${standardisedFolderPrefix}/${invoiceFileName.slice(0, 27)}.txt`,
+    });
+  });
+});
+
+export const assertQueryResultWithTestData = async (
+  { billingQty, transactionQty, eventTime }: TestData,
+  unitPrice: number,
+  vendorId: string,
+  serviceName: string
+): Promise<void> => {
+  const tableName = TableNames.BILLING_TRANSACTION_CURATED;
+  const response: BillingTransactionCurated =
+    await queryResponseFilterByVendorServiceNameYearMonth(
+      vendorId,
+      serviceName,
+      tableName,
+      eventTime
+    );
+
+  const billingPrice = unitPrice * billingQty;
+  const transactionPrice = unitPrice * transactionQty;
+  const qtyDifference = billingQty - transactionQty;
+  const priceDifference = billingPrice - transactionPrice;
+  const priceDifferencePercentage = (priceDifference / transactionPrice) * 100;
+  const qtyDifferencePercentage = (qtyDifference / transactionQty) * 100;
+
+  if (billingQty !== undefined && transactionQty !== undefined) {
+    expect(response[0].billing_quantity).toEqual(billingQty);
+    expect(response[0].transaction_quantity).toEqual(transactionQty);
+    expect(response[0].quantity_difference).toEqual(String(qtyDifference));
+    expect(response[0].quantity_difference_percentage).toEqual(
+      String(qtyDifferencePercentage)
+    );
+    expect(response[0].transaction_price).toEqual(transactionPrice.toFixed(4));
+    expect(response[0].price_difference).toEqual(priceDifference.toFixed(4));
+    expect(response[0].billing_price).toEqual(billingPrice.toFixed(4));
+    expect(response[0].price_difference_percentage).toEqual(
+      priceDifferencePercentage.toFixed(4)
+    );
+  } else if (billingQty !== undefined && transactionQty === undefined) {
+    expect(response[0].billing_quantity).toEqual(billingQty);
+    expect(response[0].billing_price).toEqual(billingPrice.toFixed(4));
+
+    expect(response[0].quantity_difference).toEqual(String(billingQty - 0));
+    expect(response[0].quantity_difference_percentage).toEqual(undefined);
+
+    expect(response[0].price_difference).toEqual((billingPrice - 0).toFixed(4));
+    expect(response[0].price_difference_percentage).toEqual(undefined);
+  } else if (transactionQty !== undefined && billingQty === undefined) {
+    expect(response[0].transaction_quantity).toEqual(transactionQty);
+    expect(response[0].transaction_price).toEqual(transactionPrice.toFixed(4));
+
+    expect(response[0].quantity_difference_percentage).toEqual(
+      String(0 - transactionQty * 100)
```

Comment:
`0 - transactionQty` is the same as just `-transactionQty`

Reply:
yes removed the 0